### DISCUSSION TITLE
Contract Testing using Pact

### DISCUSSION THREAD

**Discussion by akshay-nm** [reduxjs/redux-toolkit#2230] [Q&A]
I am trying to setup contract tests for my react application. 
For this I'm trying to send api calls using the `enpoint.initiate()` way. After that, I test the `action`.
I have the pact mock ready. It simulates the backend.
I also have the test code ready. 
First time it ran, it failed saying I need to pass a custom fetchFn for SSR. 

I've tried importing isomorphic-fetch, node-fetch, cross-fetch in my setup file for jest.
But now it either says `Only absolute URLs supported` or `ERRCONNREFUSED: 127.0.0.1:80`.

I don't know why it's trying to reach port 80. I have the app configured to hit port 7000 for API. 
The applications works when I run it in the browser and hits the API.

I don't know what I am missing here. Any suggestions?

More on pact: https://docs.pact.io/

I thought setting tests would be fun and easy. Why is it so difficult to send HTTP requests in jest tests? 😭

**Comment by phryneas** [comment]
Well, *are* you using an absolute `baseUrl`, so something that starts with `http://` or `https://`?

**Reply by akshay-nm** [reply]
Yes, I am using `http://localhost:7000`. 
But it was specified in `.env.local`, which is ignored by jest (create-react-app). Creating a copy named `.env.test.local` fixed it! 
https://github.com/facebook/create-react-app/issues/5325#issuecomment-427577882

Thank you @phryneas for pointing me in the right direction.
 I guess all I needed was a good night's sleep and a new perspective. 😅
